odoo/models.py

```diff
@@ -2072,16 +2069,87 @@ def _read_group_fill_temporal(self, data, groupby, aggregated_fields, annotated_
         if first_a_gby['type'] not in ('date', 'datetime'):
             return data
         interval = first_a_gby['interval']
+        granularity = first_a_gby['granularity']
         groupby_name = groupby[0]
 
         # existing non null datetimes
-        existing = [d[groupby_name] for d in data if d[groupby_name]]
-
-        if len(existing) < 2:
+        existing = sorted([d[groupby_name] for d in data if d[groupby_name]])
+        existing_from = existing[0] if existing else False
+        existing_to = existing[-1] if existing else False
+
+        # fill temporal between bounds
+        # the context keys : fill_temporal_from and fill_temporal_to offer a way to guarantee
+        # that at least a certain date range will be returned as groups. They are also the
+        # bounds to fill. Any group outside those bounds will not be removed, but there will
+        # be no empty group added. If any of those keys is omitted, existing bounds are used,
+        # if applicable. If fill_temporal_from is not specified, and there is no data, no group
+        # is returned.
+        # edge case example:
+        # existing:           [Jan, Sep]
+        # fill_temporal_from: [Apr]
+        # fill_temporal_to:   [Jul]
+        # result:             [Jan, |Apr, May, Jun, Jul|, Sep]
+        #
+        # general usage examples:
+        # 1. We have a Forecast in a Kanban view. We are in April and we want the
+        # forecast to cover the [Apr - Jul] period.
+        #     - domain:             [(>= Apr), (<= Jul)] (limit the read_group to the forecast)
+        #     - fill_temporal_from: Apr (current month)
+        #     - fill_temporal_to:   Jul
+        # The result would be [Apr, May, Jun, Jul]
+        #
+        # 2. We want the same Forecast [Apr - Jul], but there is a chance that some
+        # records were forgotten in Jan, and we want to be able to reschedule
+        # them, but we don't need the fill_temporal to take effect from Jan.
+        #     - domain:             [(<= Jul)] (we removed the first bound)
+        #     - fill_temporal_from: Apr (current month)
+        #     - fill_temporal_to:   Jul
+        # The result would be [Jan, |Apr, May, Jun, Jul|]. We have the forecast
+        # with the fill_temporal, and every month preceding the forecast as long
+        # as there is at least one record in it (no empty month before the
+        # forecast).
+        tz = next((date.tzinfo for date in existing if date.tzinfo), False)
+        fill_from = self.env.context.get('fill_temporal_from')
+        fill_to = self.env.context.get('fill_temporal_to')
```

Comment:
I thought about the object too, but we would have to be very cautious about overwriting the `Object` with `True`, since in this case, we would lose the filling details. 
But since fill_temporal usage isn't wide-spread at the moment, I could make this work :p

Reply:
I wouldn't expect that to be much of an issue, I'd think it's the view requesting that the timeline be filled which would either just set it to `true` (as currently), or set it to something more complicated.

The only big question is what should happen if the key is set to an empty dict / object: does that correspond to `True` (enable with default behaviour) or `False` (don't enable)?

Reply:
That seems extremely bleh, and no better.

Reply:
So bleh it. Then I'll pick `True` cuz it'll probably win the race (it seems less likely that the object would be evaluated more often in backend)